_20240220_Generics/src/task02/medziai/Kadagys.java

```diff
@@ -0,0 +1,8 @@
+package task02.medziai;
+
+public class Kadagys extends Spygliuotis{
+    @Override
+    public String turi() {
```

Comment:
![image](https://github.com/netoks/CATasks/assets/36195315/437b6af1-848e-42dc-bb5d-8b595037f607)
turi() funkcija neturi kartotis visose klasėse:)

Reply:
nesu tikras kaip šitoj vietoj tuomet Kadagys klaseje neperrašant metodo atspaudinti "Kadagys turi spyglius".. Spygliuotis negali žinoti koks konkretus medis spausdina..

Reply:
gali:)

Jei naudojam

Spygliuotis kadagys = new Kadagys();
kadagys.turi();

turi() metode žinosime, kad jį kviečia kadagys tipo objektas (this):)

Reply:
šitoj vietoj man reikės pagalbos, nesugalvoju kaip išpręsti :) aš tėvynei klasei negaliu perduoti, kokį konkretų medžio pavadinimą atspausdinti su turi() metodu. nebent Lapuotis/Spygliuotis yra turi() metodas, o konkretūs medžiai toString() metodą perrašo panaudojant turi(), bet tuomet jau nebeatitinka užduoties..

Reply:
Spygliuotis klasėje turim metodą:

```
    public void turi() {
        System.out.println(this + " turi spyglius");
    }
```

Egle.java

turim metodą:
```
    @Override
    public String toString() {
        return "Eglė";
    }
```

Kviečiam:
```
    Egle egle = new Egle();
    egle.turi();
```

Rezultatas:
```
Egle turi spyglius

```

Reply:
sutvarkyta  :)